Export to_onnx fails when model has multiple inputs

## 🐛 Bug

Export to_onnx fails when model has multiple inputs and example output is not provided

```
    return func(*args, **kwargs)
pytorch_lightning/core/lightning.py:1896: in to_onnx
    kwargs["example_outputs"] = self(input_sample)

...

    def _call_impl(self, *input, **kwargs):
        forward_call = (self._slow_forward if torch._C._get_tracing_state() else self.forward)
        # If we don't have any hooks, we want to skip the rest of the logic in
        # this function, and just call forward.
        if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
                or _global_forward_hooks or _global_forward_pre_hooks):
>           return forward_call(*input, **kwargs)
E           TypeError: forward() missing 1 required positional argument: 'y'
```

### To Reproduce

Can be reproduced using a model from test_model_summary.py

```
class UnorderedModel(LightningModule):
    """A model in which the layers not defined in order of execution"""

    def __init__(self):
        super().__init__()
        # note: the definition order is intentionally scrambled for this test
        self.layer2 = nn.Linear(10, 2)
        self.combine = nn.Linear(7, 9)
        self.layer1 = nn.Linear(3, 5)
        self.relu = nn.ReLU()
        # this layer is unused, therefore input-/output shapes are unknown
        self.unused = nn.Conv2d(1, 1, 1)

        self.example_input_array = (torch.rand(2, 3), torch.rand(2, 10))

    def forward(self, x, y):
        out1 = self.layer1(x)
        out2 = self.layer2(y)
        out = self.relu(torch.cat((out1, out2), 1))
        out = self.combine(out)
        return out

model = UnorderedModel()
model.to_onnx(file_path)

```

### Expected behavior

Export is successful

### Environment

```
* CUDA:
	- GPU:
		- NVIDIA GeForce 940MX
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 20210808
	- tqdm:              4.43.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.6.14
	- version:           #1 SMP Sun Jul 25 16:24:19 UTC 2021
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -153,6 +153,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed truncated backprop through time enablement when set as a property on the LightningModule and not the Trainer ([#8804](https://github.com/PyTorchLightning/pytorch-lightning/pull/8804/))
 
 
+- Fixed an issues with export to ONNX format when a model has multiple inputs ([#8800](https://github.com/PyTorchLightning/pytorch-lightning/pull/8800))
+
+
 ## [1.4.0] - 2021-07-27
 
 ### Added

--- a/pytorch_lightning/core/lightning.py
+++ b/pytorch_lightning/core/lightning.py
@@ -1891,7 +1891,10 @@ class LightningModule(
 
         if "example_outputs" not in kwargs:
             self.eval()
-            kwargs["example_outputs"] = self(input_sample)
+            if isinstance(input_sample, Tuple):
+                kwargs["example_outputs"] = self(*input_sample)
+            else:
+                kwargs["example_outputs"] = self(input_sample)
 
         torch.onnx.export(self, input_sample, file_path, **kwargs)
         self.train(mode)

--- a/tests/models/test_onnx.py
+++ b/tests/models/test_onnx.py
@@ -23,6 +23,7 @@ import tests.helpers.utils as tutils
 from pytorch_lightning import Trainer
 from tests.helpers import BoringModel
 from tests.helpers.runif import RunIf
+from tests.utilities.test_model_summary import UnorderedModel
 
 
 def test_model_saves_with_input_sample(tmpdir):
@@ -66,10 +67,17 @@ def test_model_saves_with_example_output(tmpdir):
     assert os.path.exists(file_path) is True
 
 
-def test_model_saves_with_example_input_array(tmpdir):
-    """Test that ONNX model saves with_example_input_array and size is greater than 3 MB"""
-    model = BoringModel()
-    model.example_input_array = torch.randn(5, 32)
+@pytest.mark.parametrize(
+    ["modelclass", "input_sample"],
+    [
+        (BoringModel, torch.randn(1, 32)),
+        (UnorderedModel, (torch.rand(2, 3), torch.rand(2, 10))),
+    ],
+)
+def test_model_saves_with_example_input_array(tmpdir, modelclass, input_sample):
+    """Test that ONNX model saves with example_input_array and size is greater than 3 MB"""
+    model = modelclass()
+    model.example_input_array = input_sample
 
     file_path = os.path.join(tmpdir, "model.onnx")
     model.to_onnx(file_path)

